### PR TITLE
feat(ui, core): Added MessageInputController and general refactor

### PR DESCRIPTION
**Pull request by deven98**

### CODE REVIEW

File: packages/stream_chat/lib/src/core/models/message.dart [hunk 1]
```diff
@@ -72,16 +72,19 @@ class Message extends Equatable {
     this.pinnedBy,
     this.extraData = const {},
     this.deletedAt,
-    this.status = MessageSendingStatus.sent,
+    this.status = MessageSendingStatus.sending,
```
Comment (HayesGordon) [inline packages/stream_chat/lib/src/core/models/message.dart:75]: Why was this changed? Note that the `fromJson` is setting `MessageSendingStatus.sent`

Comment (imtoori) [reply packages/stream_chat/lib/src/core/models/message.dart:75]: because we need that in order to have the right behaviour in the message input
a new created message has a sending status, while if you get it from backend we set it as send
